client/src/main/com/sinch/sdk/domains/sms/models/BatchText.java

```diff
@@ -9,12 +9,12 @@
  * @since 1.0
  */
 public class BatchText extends Batch<String> {
-  private final boolean flashMessage;
+  private final Boolean flashMessage;
   private final Parameters parameters;
-  private final boolean truncateConcat;
-  private final int maxNumberOfMessageParts;
-  private final int fromTon;
-  private final int fromNpi;
+  private final Boolean truncateConcat;
```

Comment:
Why do you need to use Objects instead of primitives?

Reply:
Due to autoboxing.
When receiving a DTO instance where fields is not present or null: autoboxing is causing NullPointerException to map missing value to primitive.
e.g.: Sending a SMS with only `to`and `body` field (see sample-app `Send` source)
REST API is returning:
```
{
    "id": "01HEAC0AG69SVYYQ675VPYT28Q",
    "to": [
        "XXXXX"
    ],
    "canceled": false,
    "body": "the body",
    "type": "mt_text",
    "created_at": "2023-11-03T10:35:03.558Z",
    "modified_at": "2023-11-03T10:35:03.558Z",
    "delivery_report": "none",
    "expire_at": "2023-11-06T10:35:03.558Z",
    "feedback_enabled": false,
    "flash_message": false
}
```
Then BatchText.truncatConcat as boolean primitive cannot contains "valid" information (nor true, nor false) and autoboxing during convertion from DTO(https://github.com/sinch/sinch-sdk-java/pull/2/files#diff-45ba5a02d31c9bfab5d5e2a1c03fdf3809bc78697f9dcbbffbc9c0bc6652f4a2R110) is throwing the exception because:
- `dto.getTruncateConcat()` is returning `null` (field is a `Boolean` and not `boolean`)
- then `null` cannot be converted to a boolean because of null value
- then NullPointerException

Instead of modifying DTO object to use primitives, I'm using Object to be able have distinction between what was recevied/set and what was not.
In addition if there is some default value to be set; then it is to the server side to set this default value on return.